frontend/src/store/reducers/surveys.js

```diff
@@ -121,14 +122,18 @@ export default function(state = initialState, action) {
     case UPDATE_QUESTION_IN_DRAFT_SURVEY:
       draft = { ...state.draft };
 
-      if (action.payload.id)
-        draft.questions.find(
-          question => question.id === action.payload.id
-        ).value = action.payload.value;
-      else if (action.payload.order)
-        draft.questions.find(
-          question => question.order === action.payload.order
-        ).value = action.payload.value;
+      if (action.payload.id) {
+        const q = draft.questions.find(
+          question => question.id === action.payload.id);
+        q.value = action.payload.value;
+        q.subject = action.payload.subject;
+      }
+      else if (action.payload.order) {
+        const q = draft.questions.find(
+          question => question.order === action.payload.order);
+        q.value = action.payload.value;
+        q.subject = action.payload.subject;
+      }
```

Comment:
Generally, redux documentation says that reducer functions should be immutable. So it would be good if you changed it to something like:
```javascript
const affectedQuestionIndex = draft.questions.findIndex(q => q.id === action.payload.id)
const newDraft = [
  ...draft.slice(0, affectedQuestionIndex), 
  { ...draft[affectedQuestionIndex], value: action.payload.value, subject: action.payload.value }, 
  ...draft.slice(affectedQuestionIndex+1)
];
return {...state, draft: newDraft}
```
Of course with `if`s ;)

Reply:
Now I noticed that it IS actually immutable, because there is 
```javascript 
const draft = {...state.draft}
```
:D So you can leave it as is